java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java

```diff
@@ -22,17 +22,20 @@
 import com.google.common.base.Preconditions;
 import com.google.common.collect.ImmutableList;
 
-import org.sonar.java.se.constraint.BooleanConstraint;
-
 import java.util.List;
 
 public abstract class BinarySymbolicValue extends SymbolicValue {
 
   SymbolicValue leftOp;
   SymbolicValue rightOp;
 
+  public BinarySymbolicValue() {
```

Comment:
All the modification of this file can go away and only the constructor with operand of RelationalSymbolicValue can be left. Operands can be initialized. (Precondition can be kept).